piet-common/src/lib.rs

```diff
@@ -29,20 +29,21 @@ pub use piet::*;
 #[doc(hidden)]
 pub use piet::kurbo;
 
-#[cfg(any(
-    feature = "cairo",
-    not(any(target_arch = "wasm32", target_os = "windows", feature = "direct2d"))
-))]
-#[path = "cairo_back.rs"]
-mod backend;
-
-#[cfg(any(feature = "d2d", all(target_os = "windows", not(feature = "cairo"))))]
-#[path = "direct2d_back.rs"]
-mod backend;
-
-#[cfg(any(feature = "web", target_arch = "wasm32"))]
-#[path = "web_back.rs"]
-mod backend;
+cfg_if::cfg_if! {
+    // if we have explicitly asked for cairo *or* we are not wasm, web, or windows:
+    if #[cfg(any(feature = "cairo", not(any(target_arch = "wasm32", feature="web", target_os = "windows"))))] {
+        #[path = "cairo_back.rs"]
+         mod backend;
```

Comment:
Whoops, didn't notice that extra formating space earlier. Fine to keep it I guess, but I just wanted to let the info be out there. Really unfortunate that rustfmt doesn't work on this.

Reply:
😆